gadget/validate.go

```diff
@@ -230,31 +240,73 @@ func ensureRolesConsistency(roles map[string]*roleInstance, expectedSeed bool) e
 	return nil
 }
 
+func ensureRolesConsistencyOnModalClassic(roles map[string]*roleInstance) error {
+	if roles[SystemBoot] == nil || roles[SystemData] == nil {
+		return fmt.Errorf("system-boot and system-data roles are needed on classic")
```

Comment:
we don't seem to have a similar check for other systems that boot is required?

Reply:
Which devices with UC20+ do not have a ubuntu-boot partition? @kubiko maybe you know? I'd like to see their gadgets.

Reply:
we decided to keep the strict check for now

Reply:
Fwiw, I agree with this decision.